java/vector/src/main/codegen/templates/HolderReaderImpl.java

```diff
@@ -131,6 +127,15 @@ public void read(Nullable${name}Holder h) {
     byte[] value = new byte [holder.byteWidth];
     holder.buffer.getBytes(0, value, 0, holder.byteWidth);
     return value;
+  <#elseif minor.class == "TimeStampSec">
+    final long millis = java.util.concurrent.TimeUnit.SECONDS.toMillis(holder.value);
+    return DateUtility.getLocalDateTimeFromEpochMilli(millis);
+  <#elseif minor.class == "TimeStampMilli" || minor.class == "DateMilli" || minor.class == "TimeMilli">
+    return DateUtility.getLocalDateTimeFromEpochMilli(holder.value);
+  <#elseif minor.class == "TimeStampMicro">
+    return DateUtility.getLocalDateTimeFromEpochMicro(holder.value);
+  <#elseif minor.class == "TimeStampNano">
+    return DateUtility.getLocalDateTimeFromEpochNano(holder.value);
```

Comment:
I think this was not right before.  Each of these classes used the general constructor below which became `new LocalDateTime(holder.value)`. Using that constructor, the value is expected to be in milliseconds, but we were passing in nano and micros in without any conversion.